sympy/solvers/inequalities.py

```diff
@@ -1007,3 +1009,156 @@ def reduce_inequalities(inequalities, symbols=[]):
+            for j in range(C.cols):
+                if C[j] < 0:
+                    piv_cols.append(j)
+
+            if not piv_cols:
+                return M, R, S
+            rand.shuffle(piv_cols)
+            j0 = piv_cols[0]
+
+            piv_rows = []
+            for i in range(A.rows):
+                if A[i, j0] > 0:
+                    ratio = B[i] / A[i, j0]
+                    piv_rows.append((ratio, i))
+
+            if not piv_rows:
+                raise UnboundedLinearProgrammingError('Objective function can assume arbitrarily large positive (resp. negative) values at feasible vectors!')
+            piv_rows = sorted(piv_rows, key=lambda x: (x[0], x[1]))
+            piv_rows = [(ratio, i) for ratio, i in piv_rows if ratio == piv_rows[0][0]]
+            rand.shuffle(piv_rows)
+            _, i0 = piv_rows[0]
+
+            M = _pivot(M, i0, j0)
+            R[j0], S[i0] = S[i0], R[j0]
+        else:
+            for k in range(B.rows):
+                if B[k] < 0:
+                    break
+
+            piv_cols = []
+            for j in range(A.cols):
+                if A[k, j] < 0:
+                    piv_cols.append(j)
+            if not piv_cols:
+                raise InfeasibleLinearProgrammingError('The constraint set is empty!')
+            rand.shuffle(piv_cols)
+            j0 = piv_cols[0]
+
+            ratio = B[k] / A[k, j0]
+            piv_rows = [(ratio, k)]
+            for i in range(A.rows):
+                if A[i, j0] > 0 and B[i] > 0:
+                    ratio = B[i] / A[i, j0]
+                    piv_rows.append((ratio, i))
+
+            piv_rows = sorted(piv_rows, key=lambda x: (x[0], x[1]))
+            piv_rows = [(ratio, i) for ratio, i in piv_rows if ratio == piv_rows[0][0]]
+            rand.shuffle(piv_rows)
+            _, i0 = piv_rows[0]
+
+            M = _pivot(M, i0, j0)
+            R[j0], S[i0] = S[i0], R[j0]
+
+
+# Maximize Cx + D constrained with Ax <= B and x >= 0
+# Minimize y^{T}B constrained with y^{T}A >= C^{T} and y >= 0
+def linear_programming(A, B, C, D = Matrix([0])):
+    """
+    When x is a column vector of variables, y is a column vector of dual variables,
+    and when the objective is either:
+
+    *) Maximizing Cx constrained to Ax <= B and x >= 0.
+    *) Minimizing y^{T}B constrained to y^{T}A >= C^{T} and y >= 0.
+
+    The method returns a triplet of solutions optimum, argmax and argmax_dual where
+    optimum is the optimum solution, argmax is x and argmax_dual is y.
+
+    Examples
+    ========
+    >>> from sympy.matrices.dense import Matrix
+    >>> from sympy.solvers.inequalities import linear_programming
+    >>> A = Matrix([[0, 1, 2], [-1, 0, -3], [2, 1, 7]])
+    >>> B = Matrix([3, -2, 5])
+    >>> C = Matrix([[1, 1, 5]])
+    >>> D = Matrix([0])
+    >>> linear_programming(A, B, C, D)
+    (11/3, [0, 1/3, 2/3], [0, 2/3, 1])
+
+    The method also works with inegers, floats, symbolic expressions (like sqrt(2)) and symbols (like x).
```

Comment:
I don't think that this should work with symbols. The reason that the code runs okay with symbols is no more coincidental that `sorted` works with any python objects, and that is not compatible with comparing the magnitude of  real numbers.
Linear inequality over symbolic coefficients should actually be comprehended as polynomial inequality that is quantified over reals, so it's not the scope of this.

Reply:
ok. 
I get what you mean.
I'll remove it then from the docstring and refactor the tests which regard symbols.